dspace-server-webapp/src/test/java/org/dspace/app/rest/signposting/controller/LinksetRestControllerIT.java

```diff
@@ -0,0 +1,677 @@
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "' " +
+                        "&& @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset')]").exists())
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "/json" +
+                        "' && @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset+json')]").exists());
+
+        DSpaceServicesFactory.getInstance().getConfigurationService().reloadConfig();
+        metadataAuthorityService.clearCache();
+        choiceAuthorityService.clearCache();
+    }
+
+    @Test
+    public void findTypedLinkForBitstreamOfWorkspaceItem() throws Exception {
+        String bitstreamContent = "ThisIsSomeDummyText";
+        String bitstreamMimeType = "text/plain";
+
+        context.turnOffAuthorisationSystem();
+        WorkspaceItem workspaceItem = WorkspaceItemBuilder.createWorkspaceItem(context, collection)
+                .withTitle("Workspace Item")
+                .build();
+        Item item = workspaceItem.getItem();
+        itemService.addMetadata(context, item, "dc", "identifier", "doi", Item.ANY, doi);
+
+        Bitstream bitstream = null;
+        try (InputStream is = IOUtils.toInputStream(bitstreamContent, CharEncoding.UTF_8)) {
+            bitstream = BitstreamBuilder.createBitstream(context, workspaceItem.getItem(), is)
+                    .withName("Bitstream")
+                    .withDescription("description")
+                    .withMimeType(bitstreamMimeType)
+                    .build();
+        }
+        context.restoreAuthSystemState();
+
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+        getClient().perform(get("/signposting/links/" + bitstream.getID())
+                        .header("Accept", "application/json"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$",
+                        Matchers.hasSize(3)))
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/items/" + item.getID() + "' " +
+                        "&& @.rel == 'collection' " +
+                        "&& @.type == 'text/html')]").exists())
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "' " +
+                        "&& @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset')]").exists())
+                .andExpect(jsonPath("$[?(@.href == '" + uiUrl + "/signposting/linksets/" + item.getID() + "/json" +
+                        "' && @.rel == 'linkset' " +
+                        "&& @.type == 'application/linkset+json')]").exists());
+
+        DSpaceServicesFactory.getInstance().getConfigurationService().reloadConfig();
+        metadataAuthorityService.clearCache();
+        choiceAuthorityService.clearCache();
+    }
+
+}
```

Comment:
We appear to be missing tests for the Author links.  Those currently would not work because of the incorrect Entity Type, but once that bug is fixed, we need to verify Author links are working.

Reply:
Tests updated